### PR TITLE
Add contributing.md document, format md with prettier

### PR DESCRIPTION
**Pull request by hunterwerlla**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the **README** document

- [ ] Local run of `npm run fullBuild` succeeds
- [ ] I have run tslint against my changes using the root tslint.yaml
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG** (CHANGELOG.md) if the change is feature related

## License

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+Contributions via pull requests are much appreciated. Before sending us a pull request, please ensure that:
+
+1. You are working against the latest source on the master branch.
+2. You check existing open, and recently merged, pull requests to make sure someone else hasn't addressed the problem already.
+3. You open an issue to discuss any significant work - we would hate for your time to be wasted.
+
+To send us a pull request, please:
+
+1. Fork the repository.
+2. Modify the source; please focus on the specific change you are contributing. Please follow the instructions in building to install the hook for prettier to make sure your code is formatted like the repo.
```
Comment (awschristou) [inline CONTRIBUTING.md]: Consider moving `Please follow the instructions in building to install the hook for prettier to make sure your code is formatted like the repo.` to a sub bullet (2a?) and reword it a little. Perhaps:
`To ensure your code formatting is consistent with the repo, please follow the steps in the **Building** section. This installs a code formatter.`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## Testing locally
+
+We are still working out a good way to do this beyond writing tests. 
+
+If you have a test VSTS server, you can upload the packaged plugin from "\_package". Keep in mind that VSTS expects you to bump the version number (located in \_versioninfo.json) every time you upload a new plugin, as well as that your pulibsher ID matches the publisher ID of the plugin already on the server.
+
+## Finding contributions to work on
+
+Looking at the existing issues is a great way to find something to contribute on. Looking at any ['help wanted'](https://github.com/aws/aws-toolkit-jetbrains/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) issues is a great place to start.
```
Comment (awschristou) [inline CONTRIBUTING.md]: Wrong repo (jetbrains)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+7. Pay attention to any automated CI failures reported in the pull request, and stay involved in the conversation.
+
+GitHub provides additional documentation on [forking a repository](https://help.github.com/articles/fork-a-repo/) and
+[creating a pull request](https://help.github.com/articles/creating-a-pull-request/).
+
+## Testing locally
+
+We are still working out a good way to do this beyond writing tests. 
+
+If you have a test VSTS server, you can upload the packaged plugin from "\_package". Keep in mind that VSTS expects you to bump the version number (located in \_versioninfo.json) every time you upload a new plugin, as well as that your pulibsher ID matches the publisher ID of the plugin already on the server.
```
Comment (awschristou) [inline CONTRIBUTING.md]: Is `\_versioninfo.json` rendering how you expect it to?

Comment (hunterwerlla) [reply CONTRIBUTING.md]: Yep, this was a change prettier made, but it renders just fine

File: CONTRIBUTING.md (hunk 2 above)
Comment (hunterwerlla) [reply CONTRIBUTING.md]: I thought I got all of these, but oops, fixed

File: CONTRIBUTING.md (hunk 1 above)
Comment (hunterwerlla) [reply CONTRIBUTING.md]: Changed the wording to 
"Please follow the instructions in **Building** to ensure your changes will be formatted properly (`npm install` installs a code formatter)"

**Pull request merged**